lib/Transforms/IPO/StructFieldCacheAnalysisImpl.h

```diff
@@ -227,15 +269,38 @@ class StructFieldAccessInfo {
         StructManager(M),
         StatCounts(StructFieldAccessManager::DebugStats::DS_MaxNumStats) {}
 
-  ~StructFieldAccessInfo() {}
+  ~StructFieldAccessInfo() {
+    for (auto &it : CallInstFieldAccessMap) {
+      delete it.second;
```

Comment:
Mentioned this elsewhere - if you have an unique_ptr, then you don't have to delete it explicitly here.